hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -131,6 +132,22 @@ public static DBCheckpoint createOmSnapshotCheckpoint(
       OMMetadataManager omMetadataManager, SnapshotInfo snapshotInfo)
       throws IOException {
     RDBStore store = (RDBStore) omMetadataManager.getStore();
+
+    final long dbLatestSequenceNumber =
+        ((RDBStore) omMetadataManager.getStore()).getDb()
+            .getLatestSequenceNumber();
+
+    final RocksDBCheckpointDiffer checkpointDiffer =
+        omMetadataManager.getStore().getRocksDBCheckpointDiffer();
+
+    // Set compaction log filename to the latest DB sequence number
+    // right before taking the RocksDB checkpoint
+    //
+    // Note it doesn't matter if sequence number hasn't increased (even though
+    // it shouldn't happen), since the writer always appends the file.
+    checkpointDiffer.setCompactionLogParentDir(store.getSnapshotsParentDir());
+    checkpointDiffer.setCompactionLogFilenameBySeqNum(dbLatestSequenceNumber);
+
```

Comment:
does the lastSnapshotCounter need to incremented here?

Reply:
In the latest commit e76a24eddefcc44ab429f2be3245e0ec252c7d5a I have added the logic to persist sequence number at the time of checkpoint (snapshot) creationm, so that it can be read during DAG reconstruction.